lucene/core/src/java/org/apache/lucene/document/FieldType.java

```diff
@@ -400,6 +405,42 @@ public VectorSimilarityFunction vectorSimilarityFunction() {
     return vectorSimilarityFunction;
   }
 
+  /**
+   * Enable tensor indexing. Each vector in the tensor has the same dimension. Different tensor
+   * values can vary in the number of vectors.
+   *
+   * @param isTensor Boolean flag indicating if the field indexes tensors
+   * @param dimension Dimension of each vector in the tensor
+   * @param encoding {@link VectorEncoding} for each tensor vector. Should be the same for all
+   *     vectors
+   * @param similarityFunction {@link VectorSimilarityFunction} Used to compare tensors during
+   *     indexing and search
+   * @param aggregation {@link Aggregation} used to aggregate similarity across multiple vectors
+   */
+  public void setTensorAttributes(
+      boolean isTensor,
+      int dimension,
+      VectorEncoding encoding,
+      VectorSimilarityFunction similarityFunction,
+      Aggregation aggregation) {
+    checkIfFrozen();
+    this.isTensor = isTensor;
+    this.vectorDimension = dimension;
+    this.vectorEncoding = Objects.requireNonNull(encoding);
+    this.vectorSimilarityFunction = Objects.requireNonNull(similarityFunction);
```

Comment:
Wondering if we might call `setVectorAttributes` here:
```suggestion
    setVectorAttributes(dimension, encoding, similarityFunction);
```